src/ulkoiset_rajapinnat/haku.clj

```diff
@@ -58,25 +57,17 @@
 (defn haku-resource [vuosi request user channel]
   (go
     (try
-    (let [kieli (<<?? (koodisto-as-channel "kieli"))
-          kausi (<<?? (koodisto-as-channel "kausi"))
-          hakutyyppi (<<?? (koodisto-as-channel "hakutyyppi"))
-          hakutapa (<<?? (koodisto-as-channel "hakutapa"))
-          haunkohdejoukko (<<?? (koodisto-as-channel "haunkohdejoukko"))
-          haunkohdejoukontarkenne (<<?? (koodisto-as-channel "haunkohdejoukontarkenne"))
-          haku (<<?? (fetch-haku vuosi))
+    (let [kieli (<? (koodisto-as-channel "kieli"))
+          kausi (<? (koodisto-as-channel "kausi"))
+          hakutyyppi (<? (koodisto-as-channel "hakutyyppi"))
+          hakutapa (<? (koodisto-as-channel "hakutapa"))
+          haunkohdejoukko (<? (koodisto-as-channel "haunkohdejoukko"))
+          haunkohdejoukontarkenne (<? (koodisto-as-channel "haunkohdejoukontarkenne"))
+          hakus (<? (fetch-haku vuosi))
```

Comment:
Noi <<?? oli siinä toivossa että kaikki resurssit aloitettaisiin hakemaan välittömästi, mutta jälkeenpäin ajatellen tuo ei taida toimia edes niin. Selkeyden vuoksi <? on ehdottomasti parempi. +1

Reply:
Ei kysymys ole selkeydestä eikä vaan siitä, että threadit loppuvat, koska core.asyncin !!-operaatiot (joita nuo full.asyncin ?? -operaatiot kutsuvat) täyttävät thread-poolin, koska ne blokkaavat. Eivätkä pyynnöt siksi edes pyöri rinnakkain.